compression/burrows_wheeler.py

```diff
@@ -0,0 +1,165 @@
+"""
+https://en.wikipedia.org/wiki/Burrows%E2%80%93Wheeler_transform
+
+The Burrows–Wheeler transform (BWT, also called block-sorting compression)
+rearranges a character string into runs of similar characters. This is useful
+for compression, since it tends to be easy to compress a string that has runs
+of repeated characters by techniques such as move-to-front transform and
+run-length encoding. More importantly, the transformation is reversible,
+without needing to store any additional data except the position of the first
+original character. The BWT is thus a "free" method of improving the efficiency
+of text compression algorithms, costing only some extra computation.
+"""
+
+
+def all_rotations(string):
+    """
+    :param str string: The string that will be rotated len(string) times.
+    :return: A list with len(string) rotations of the parameter string.
+    :rtype: list[str]
+    :raises TypeError: If the string parameter type is not str.
+    Examples:
+
+    >>> all_rotations("^BANANA|")
+    ['^BANANA|', 'BANANA|^', 'ANANA|^B', 'NANA|^BA', 'ANA|^BAN', 'NA|^BANA',\
+ 'A|^BANAN', '|^BANANA']
+    >>> all_rotations("a_asa_da_casa")
+    ['a_asa_da_casa', '_asa_da_casaa', 'asa_da_casaa_', 'sa_da_casaa_a',\
+ 'a_da_casaa_as', '_da_casaa_asa', 'da_casaa_asa_', 'a_casaa_asa_d',\
+ '_casaa_asa_da', 'casaa_asa_da_', 'asaa_asa_da_c', 'saa_asa_da_ca',\
+ 'aa_asa_da_cas']
+    >>> all_rotations("panamabanana")
+    ['panamabanana', 'anamabananap', 'namabananapa', 'amabananapan',\
+ 'mabananapana', 'abananapanam', 'bananapanama', 'ananapanamab',\
+ 'nanapanamaba', 'anapanamaban', 'napanamabana', 'apanamabanan']
+    >>> all_rotations(5)
+    Traceback (most recent call last):
+        ...
+    TypeError: The parameter string type must be str.
+    """
+    if not (type(string) is str):
```

Comment:
PEP8 says
> Object type comparisons should always use __isinstance()__ instead of comparing types directly.

[__flake8 E721__](https://lintlyci.github.io/Flake8Rules) will find these automatically.